pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -30,63 +40,175 @@ func (m *Metadata) AccountID(ctx context.Context) (string, error) {
+	var instance resourcecontrollerv2.ResourceInstance
+	var zonesService *zonesv1.ZonesV1
+	var listZonesOptions *zonesv1.ListZonesOptions
+	var listZonesResponse *zonesv1.ListZonesResp
+
+	bxSession, err = bxsession.New(&bluemix.Config{
+		BluemixAPIKey:         APIKey,
+		TokenProviderEndpoint: &tokenProviderEndpoint,
+		Debug:                 false,
+	})
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: bxsession.New: %v", err)
+	}
+	tokenRefresher, err = authentication.NewIAMAuthRepository(bxSession.Config, &rest.Client{
+		DefaultHeader: gohttp.Header{
+			"User-Agent": []string{http.UserAgent()},
+		},
+	})
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: authentication.NewIAMAuthRepository: %v", err)
+	}
+	err = tokenRefresher.AuthenticateAPIKey(bxSession.Config.BluemixAPIKey)
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: tokenRefresher.AuthenticateAPIKey: %v", err)
+	}
+	authenticator = &core.IamAuthenticator{
+		ApiKey: APIKey,
+	}
+	err = authenticator.Validate()
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: authenticator.Validate: %v", err)
+	}
+	// Instantiate the service with an API key based IAM authenticator
+	controllerSvc, err = resourcecontrollerv2.NewResourceControllerV2(&resourcecontrollerv2.ResourceControllerV2Options{
+		Authenticator: authenticator,
+		ServiceName:   "cloud-object-storage",
+		URL:           "https://resource-controller.cloud.ibm.com",
+	})
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: creating ControllerV2 Service: %v", err)
+	}
+	listInstanceOptions = controllerSvc.NewListResourceInstancesOptions()
+	listInstanceOptions.SetResourceID(cisServiceID)
+	listResourceInstancesResponse, _, err = controllerSvc.ListResourceInstances(listInstanceOptions)
+	if err != nil {
+		return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: ListResourceInstances: %v", err)
+	}
+	for _, instance = range listResourceInstancesResponse.Resources {
+		authenticator = &core.IamAuthenticator{
+			ApiKey: APIKey,
+		}
+
+		err = authenticator.Validate()
 		if err != nil {
-			return "", err
 		}
 
-		for _, z := range zones {
-			if z.Name == m.BaseDomain {
-				m.SetCISInstanceCRN(z.CISInstanceCRN)
-				return m.cisInstanceCRN, nil
+		zonesService, err = zonesv1.NewZonesV1(&zonesv1.ZonesV1Options{
+			Authenticator: authenticator,
+			Crn:           instance.CRN,
+		})
+		if err != nil {
+			return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: NewZonesV1: %v", err)
+		}
+		listZonesOptions = zonesService.NewListZonesOptions()
+		listZonesResponse, _, err = zonesService.ListZones(listZonesOptions)
+		if listZonesResponse == nil {
+			return CISInstanceCRN, fmt.Errorf("getCISInstanceCRN: ListZones: %v", err)
+		}
+		for _, zone := range listZonesResponse.Result {
+			if *zone.Status == "active" {
+				if *zone.Name == BaseDomain {
+					CISInstanceCRN = *instance.CRN
```

Comment:
just noticing that the metadata's CRN isn't being set any more here like it was before. is that handled somewhere else i'm missing now?

Reply:
The old code created a DNSZoneResponse structure with "CISInstanceCRN:  *instance.CRN,"